public/app/features/dashboard-scene/scene/DashboardScene.tsx

```diff
@@ -257,7 +259,7 @@ export class DashboardScene extends SceneObjectBase<DashboardSceneState> {
     this._initialUrlState = locationService.getLocation();
 
     // Switch to edit mode
-    this.setState({ isEditing: true });
+    this.setState({ isEditing: true, showHiddenElements: true });
```

Comment:
I might be confused here a bit, but would it not be better to not tie `showHiddenElements` to `isEditing` in this way? So that we don't need to search for all the places where `isEditing` changes. Just leave it be set by `onToggleHiddenElements` and only take the flag into account if `isEditing` is true? WDYT?

Reply:
`showHiddenElements` is not necessarily tide to `isEditing` here but to the enter/exit edit mode behaviors. Rationale is that we always want hidden elements to be set to `true` when entering edit mode and unset it to `false` on exit.